### PR TITLE
Fixed a race condition in peer shutdown.

### PR DESCRIPTION
**Pull request by Piotrek321**

### CODE REVIEW

File: src/blockchain/node/manager/Manager.cpp [hunk 1]
```diff
@@ -340,7 +340,7 @@ Base::Base(
         try {
             const auto boost = boost::asio::ip::make_address(addr);
 
-            if (!boost.is_v4()) {
+            if (false == boost.is_v4()) {
```
Comment (wiesiekpap) [inline src/blockchain/node/manager/Manager.cpp:343]: why are we restoring this c-era nonsense? The guidelines someone uses to justify the huge effort expended for making everything const expressly warn against unnecessary comparisons with bool values.

File: src/blockchain/node/manager/Manager.cpp [hunk 2]
```diff
@@ -1374,5 +1376,6 @@ auto Base::Wallet() const noexcept -> const node::Wallet& { return wallet_; }
 Base::~Base()
 {
     protect_shutdown([this] { shut_down(); });
+    shutdown_timers();
```
Comment (wiesiekpap) [inline src/blockchain/node/manager/Manager.cpp]: shut_down() may, and should be called earlier, so please move thistimers shutdown to inside shut_down()

**Review by wiesiekpap — CHANGES_REQUESTED** [review]
I have a problem with restoring the if(false == boolean) fallacy after it has been taken out.
However, the important bit to correct here is to put the timers shutdown inside shut_down so it is performed every time shut_down gets called.

File: src/blockchain/node/manager/Manager.cpp (hunk 1 above)
Comment (radexx99) [reply src/blockchain/node/manager/Manager.cpp:343]: We Both Agreed on this as this false prevents from future errors and all merges whas picked with that change. 
What would be the benefit on not introducing it besides the disadv of future merges ?

Comment (Piotrek321) [reply src/blockchain/node/manager/Manager.cpp:343]: hold up:D
"this false prevents from future errors " - this has nothing to do with errors. It just visual thing.

Comment (radexx99) [reply src/blockchain/node/manager/Manager.cpp:343]: yes its visual, till then someone try to chage it. we can do false = dupa or dupa = false. but im not so expert to insist this. Justus started and we do not have any significant argument to stop it. Or maybe im missing sth, would be glad to learn sth

Comment (wiesiekpap) [reply src/blockchain/node/manager/Manager.cpp:343]: With an assignment written where a boolean condition is expected, any half decent compiler from about 2005 onward issues a warning - hope you have learned something, after all.

**Review by wiesiekpap — APPROVED** [review]

**Pull request merged**